### PR TITLE
feat(commercetools): use inventory

### PR DESCRIPTION
**Pull request by alefbarbeli**
## Description

We added support for Commercetools Inventory API, wich is required for Click and Collect feature

## Related Issue
[EN-412 Use Inventory](https://vsf.atlassian.net/browse/EN-412)

## Motivation and Context
The api method `getInventory` was not implemented until now.
It is necessary because we where using the "Product availability" property for stock status, and Commercetools docs says that this prop it is not consistent.

## How Has This Been Tested?
Added `entries` property to Product page, wich is the result of query and fully functional.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have read the **CONTRIBUTING** document.

#### Changelog
- [x] I have updated the Changelog ([V1](https://github.com/DivanteLtd/vue-storefront/blob/develop/CHANGELOG.md)) [v2](https://docs-next.vuestorefront.io/contributing/creating-changelog.html) and mentioned all breaking changes in the public API.
- [ ] I have documented all new public APIs and made changes to existing docs mentioning the parts I've changed so they're up to date.

#### Tests
- [ ] I have written test cases for my code
- [x] I have tested my Pull Request on production build and (to my knowledge) it works without any issues
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

> I tested manually my code, and it works well with both:
- [ ] Default Theme
- [ ] Capybara Theme

#### Code standards
- [x] My code follows the code style of this project.

- [x] I have followed [naming conventions](https://github.com/kettanaito/naming-cheatsheet)

#### Docs
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: packages/commercetools/api-client/src/types/Api.ts [hunk 1]
```diff
@@ -25,7 +25,7 @@ import {
   Me,
   CartQueryInterface,
   CustomerPasswordToken,
-  StoreQueryResult
+  StoreQueryResult, InventoryEntryQueryResult
```
Comment (dawid-ziobro) [inline packages/commercetools/api-client/src/types/Api.ts:28]: Please move it to next line

File: packages/commercetools/composables/src/factories/useInventoryFactory.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  Context, CustomQuery,
+  FactoryParams,
+  Logger, PlatformApi,
+  sharedRef
+} from '@vue-storefront/core';
+
+export interface UseInventory<INVENTORY> {
+  entries: ComputedProperty<INVENTORY[]>;
+  inventory_entry: ComputedProperty<INVENTORY>;
+  load: (context: Context, params: { id: string } & { customQuery?: CustomQuery }) => Promise<void>;
```
Comment (dawid-ziobro) [inline packages/commercetools/composables/src/factories/useInventoryFactory.ts:13]: Should it return `void` in Promise?

Comment (dawid-ziobro) [reply packages/commercetools/composables/src/factories/useInventoryFactory.ts:13]: I don't see usage of `customQuery`. Where is it used? Also affects `search` types.

File: packages/commercetools/theme/pages/Product.vue [hunk 3]
```diff
@@ -290,6 +296,8 @@ export default {
     };
 
     return {
+      searchInventory,
+      entries,
```
Comment (dawid-ziobro) [inline packages/commercetools/theme/pages/Product.vue:300]: Is this used somewhere?

File: packages/commercetools/api-client/src/types/Api.ts (hunk 1 above)
Comment (alefbarbeli) [reply packages/commercetools/api-client/src/types/Api.ts:29]: done!

**Pull request closed without merging**